### ISSUE TITLE
[ANDROID] Fatal Exception and app crashes on order

### ISSUE THREAD

**Issue by dharaneesh** [j3k0/cordova-plugin-purchase#633]
### Error INFO
```
Fatal Exception: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=10001, result=0, data=Intent { (has extras) }} to activity {com.xxx.xxx/com.xxx.xxx.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.smartmobilesoftware.util.IabHelper.handleActivityResult(int, int, android.content.Intent)' on a null object reference
       at android.app.ActivityThread.deliverResults(ActivityThread.java:3720)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:3763)
       at android.app.ActivityThread.-wrap16(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1403)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5443)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.smartmobilesoftware.util.IabHelper.handleActivityResult(int, int, android.content.Intent)' on a null object reference
       at com.smartmobilesoftware.inappbilling.InAppBillingPlugin.onActivityResult(InAppBillingPlugin.java:453)
       at org.apache.cordova.CordovaInterfaceImpl.onActivityResult(CordovaInterfaceImpl.java:152)
       at org.apache.cordova.CordovaActivity.onActivityResult(CordovaActivity.java:358)
       at android.app.Activity.dispatchActivityResult(Activity.java:6442)
       at android.app.ActivityThread.deliverResults(ActivityThread.java:3716)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:3763)
       at android.app.ActivityThread.-wrap16(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1403)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5443)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
```

Cordova version: 7.1.0
Cordova-Android  6.3.0
Device: Android 6.0
Plugin Version: 7.0.2

### Steps to reproduce
On buy product invoking the below methods and on confirm , the payment get successful but app crashes.
```javascript
store.order(product_id);
store.when(product_id).approved(function (product) {});
```

**Comment by Dexus** [comment]
And you also try to finish the product on approved?

Von meinem iPhone gesendet

**Comment by dharaneesh** [comment]
Yes and also the payment is succesful and on closing app is crashed

 store.order(product_id);
        store.when(product_id).approved(function (product) {
        
          console.log("Purchase Successful");
          product.finish();
  
        });

**Comment by Dexus** [comment]
Please provide some Debug logs, from your App, before it crash. 
I can't reproduce this error, yet.

**Comment by mustcan** [comment]
There is a case like this. 

store.when(product_id).approved(function (product) {
    window.location.reload(true)
     order.finish();
});

with this code, android app gets refreshed continuously and causes andriod to crash eventually.

**Comment by Dexus** [comment]
you can't reload before finish that's right...

**Comment by Dexus** [comment]
This should be fixed with #713
